include/libtorrent/aux_/session_settings.hpp

```diff
@@ -77,7 +77,7 @@ namespace libtorrent { namespace aux
 		{
 			TORRENT_ASSERT((name & settings_pack::type_mask) == type);
 			if ((name & settings_pack::type_mask) != type) return;
-			int const index = name & settings_pack::index_mask;
+			size_t const index = name & settings_pack::index_mask;
```

Comment:
was there a warning here? is the type of the enum unsigned? if not, I don't see how the result here would become unsigned